crates/services/gas_price_service/src/v1/uninitialized_task/fuel_storage_unrecorded_blocks.rs

```diff
@@ -0,0 +1,123 @@
+use crate::common::fuel_core_storage_adapter::storage::{
+    GasPriceColumn,
+    UnrecordedBlocksTable,
+};
+use fuel_core_storage::{
+    kv_store::{
+        KeyValueInspect,
+        KeyValueMutate,
+    },
+    transactional::{
+        Modifiable,
+        WriteTransaction,
+    },
+    Error as StorageError,
+    StorageAsMut,
+    StorageAsRef,
+    StorageMutate,
+};
+use fuel_core_types::fuel_merkle::storage::StorageMutateInfallible;
+use fuel_gas_price_algorithm::{
+    v1,
+    v1::UnrecordedBlocks,
+};
+
+pub trait AsUnrecordedBlocks {
+    type Wrapper<'a>: UnrecordedBlocks
+    where
+        Self: 'a;
+
+    fn as_unrecorded_blocks(&mut self) -> Self::Wrapper<'_>;
+}
```

Comment:
Nice wrapper!

Reply:
It was @xgreenx 's I had a newtype to solve the orphan rule problem before and Green switch it to this in his suggestions.